internal/commands/crypto.go

```diff
@@ -246,17 +250,45 @@ func newCryptoGenerateCmd(ctx *CmdCtx, category, algorithm string) (cmd *cobra.C
 // CryptoRandRunE is the RunE for the authelia crypto rand command.
 func (ctx *CmdCtx) CryptoRandRunE(cmd *cobra.Command, args []string) (err error) {
 	var (
-		random string
+		random     string
+		quiet      bool
+		outputPath string
 	)
 
 	if random, err = flagsGetRandomCharacters(cmd.Flags(), cmdFlagNameLength, cmdFlagNameCharSet, cmdFlagNameCharacters); err != nil {
 		return err
 	}
 
-	fmt.Printf("Random Value: %s\n", random)
+	quiet, _ = cmd.Flags().GetBool(cmdFlagNameQuiet)
 
-	if value := url.QueryEscape(random); random != value {
-		fmt.Printf("Random Value (URL Encoded): %s\n", value)
+	if outputPath, err = flagsGetFilePath(cmd.Flags(), cmdFlagNameOutputFile, cmdFlagNameDirectory); err != nil {
+		return fmt.Errorf("error getting output file path: %v", err)
+	}
+
+	if outputPath != "" {
+		dir := filepath.Dir(outputPath)
+
+		if err = os.MkdirAll(dir, 0755); err != nil {
```

Comment:
Not sure if creating all parent directories is an expected or common behavior in CLI programs. I think we should rather error when the output file parent directory doesn't exist by just using `os.WriteFile` and nothing else.